client/ipa-getkeytab.c

```diff
@@ -221,21 +274,26 @@ static int ipa_ldap_bind(const char *server_name, krb5_principal bind_princ,
                                &bv, NULL, NULL, NULL);
         if (ret != LDAP_SUCCESS) {
             fprintf(stderr, _("Simple bind failed\n"));
-            goto done;
         }
     } else {
-        ret = ldap_sasl_interactive_bind_s(ld, NULL, "GSSAPI",
-                                           NULL, NULL, LDAP_SASL_QUIET,
-                                           ldap_sasl_interact, bind_princ);
+        if (strcmp(mech, LDAP_SASL_EXTERNAL) == 0) {
+            ret = ldap_sasl_bind_s(ld, NULL, LDAP_SASL_EXTERNAL,
+                               NULL, NULL, NULL, NULL);
+        } else {
+            ret = ldap_sasl_interactive_bind_s(ld, NULL, LDAP_SASL_GSSAPI,
+                                               NULL, NULL, LDAP_SASL_QUIET,
+                                               ldap_sasl_interact, bind_princ);
+        }
+
         if (ret != LDAP_SUCCESS) {
 #ifdef LDAP_OPT_DIAGNOSTIC_MESSAGE
             ldap_get_option(ld, LDAP_OPT_DIAGNOSTIC_MESSAGE, (void*)&msg);
 #endif
             fprintf(stderr, "SASL Bind failed %s (%d) %s!\n",
                             ldap_err2string(ret), ret, msg ? msg : "");
             goto done;
+            }
         }
```

Comment:
please make sure the close parens matches the same indentation of the open one at all times